docs/configuration.md

```diff
@@ -2,72 +2,68 @@
 
-## Floating IPs
+## Floating IP
 
-You have to be able to at least assign floating IPs in your OpenStack. If you don't have rights 
-to create floating IPs you have to make sure they already exist before creating the cluster. You can create one using,
+You have to be able to create a floating IP in your OpenStack. You can create one using,
 
 ```bash
 openstack floating ip create <public network>
 ```
 The IP must be exposed as an environment variable `OPENSTACK_CONTROLPLANE_IP`.
 
-## Operating System Images
-
-We currently depend on an update version of cloud-init otherwise the operating system choice is yours. The kubeadm bootstrap provider we're using also depends on some pre-installed software like a controller-runtime, kubelet, kubeadm, etc.. . For an examples how to build such an image take a look at [image-builder (openstack)](https://image-builder.sigs.k8s.io/capi/providers/openstack.html).
+Note: Only user with admin role can create a floating IP with specific IP.
 
-The image can be referenced by exposing it as an environment variable `OPENSTACK_IMAGE_NAME`.
+## SSH authorized key
 
-## SSH Authorized Key
+The ssh public key is required. This key does not need to be created by OpenStack key pair.
 
-Openstack ssh keypair is required for the setup. This can be generated using command,
+The public key must be exposed as an environment variable `OPENSTACK_SSH_AUTHORIZED_KEY`.
 
-```bash
-openstack keypair create <name>
-```
-The keypair must be exposed as an environment variable `OPENSTACK_SSH_AUTHORIZED_KEY`.
-
-If you want to login to each machine by ssh, you have to configure security groups. If `spec.managedSecurityGroups` of `OpenStackCluster` set to true, two security groups will be created. One is `k8s-cluster-${NAMESPACE}-${CLUSTER_NAME}-secgroup-controlplane`, another is `k8s-cluster-${NAMESPACE}-${CLUSTER_NAME}-secgroup-worker`. These security group rules are following kubeadm's [Check required ports](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/install-kubeadm/#check-required-ports) so that each node can not be logged in through ssh by default. Please add existing security group allowing ssh port to `OpenStackMachineTemplate` spec. Here is an example:
+If you want to login to each machine by ssh, you have to configure security groups. If `spec.managedSecurityGroups` of `OpenStackCluster` set to true, two security groups will be created and added to the instances. One is `k8s-cluster-${NAMESPACE}-${CLUSTER_NAME}-secgroup-controlplane`, another is `k8s-cluster-${NAMESPACE}-${CLUSTER_NAME}-secgroup-worker`. These security group rules include the kubeadm's [Check required ports](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/install-kubeadm/#check-required-ports) so that each node can not be logged in through ssh by default. Please add existing security group allowing ssh port to OpenStackMachineTemplate spec. Here is an example:
```

Comment:
```suggestion
If you want to login to each machine by ssh, you have to configure security groups. If `spec.managedSecurityGroups` of `OpenStackCluster` set to true, two security groups will be created and added to the instances. One is `k8s-cluster-${NAMESPACE}-${CLUSTER_NAME}-secgroup-controlplane`, another is `k8s-cluster-${NAMESPACE}-${CLUSTER_NAME}-secgroup-worker`. These security group rules include the kubeadm's [Check required ports](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/install-kubeadm/#check-required-ports) so that each node can not be logged in through ssh by default. Please add pre-existing security group allowing ssh port to OpenStackMachineTemplate spec. Here is an example:
```